wsproto/utf8validator.py

```diff
@@ -0,0 +1,227 @@
+###############################################################################
+#
+# The MIT License (MIT)
+#
+# Copyright (c) Crossbar.io Technologies GmbH
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+###############################################################################
+
+# Note: This code is a Python implementation of the algorithm
+# "Flexible and Economical UTF-8 Decoder" by Bjoern Hoehrmann
+# bjoern@hoehrmann.de, http://bjoern.hoehrmann.de/utf-8/decoder/dfa/
```

Comment:
Ok, looking again, maybe it's not so bad – the [uncovered code that was bothering me](https://codecov.io/gh/python-hyper/wsproto/src/rewrite-the-third/wsproto/utf8validator.py) appears to be (a) the python 3 version that we never use, and (b) the `decode` method that we never use. The py2 `validate` method that we care about is actually covered. I'm not really sure how meaningful that is given that the relevant state space is this opaque DFA thing that coverage can't see, but we know the autobahn protocol tests do exercise some reasonable amount of weird utf8 edge cases (certainly enough to have alerted us to the problem in the first place!).

It does seem like we might want to chop out the code we aren't using to fix the `six` issue and make codecov happier?